http/codegen/example_server.go

```diff
@@ -256,6 +256,19 @@ func handleHTTPServer(ctx context.Context, u *url.URL{{ range $.Services }}{{ if
 		{{ .Service.VarName }}Server = {{ .Service.PkgName }}svr.New(nil, mux, dec, enc, eh, nil)
 		{{-  end }}
 	{{- end }}
+	{{- if .Services }}
+		if debug {
+			for _, server := range []interface {
+				Use(func(http.Handler) http.Handler)
+			}{
+				{{- range $svc := .Services }}
+				{{ .Service.VarName }}Server,
+				{{- end }}
+			} {
+				server.Use(httpmdlwr.Debug(mux, os.Stdout))
+			}
+		}
+	{{- end }}
```

Comment:
It might be better to define an interface like this:

```go
type (
	Server interface {
		Use(func(http.Handler) http.Handler)
	}

	Servers []Server
)

func (s Server) Use(m func(http.Handler) http.Handler) {
	for _, v := range s {
		v.Use(m)
	}
}

Reply:
Yes that makes sense (I guess you meant `func (s Servers)`), would be a lot cleaner.